### PR TITLE
python -m web_poet.testing rerun

### PR DESCRIPTION
**Pull request by kmike**
A command to print the current page object output, given the inputs stored in a fixture. It enables this very simplistic workflow: 

1. change a page object so that its output is correct (e.g. implement extraction for a field);
2. `python -m web_poet.testing rerun fixtures/pages.mywebsite.MyProductPage/test-1`;
3. copy relevant parts of the output, paste to the output.json file.

I think that even if we implement something which supports the TDD workflow better, a simple command to re-run the page object can still be useful.

TODO:

* [x] tests
* [x] docs

### CODE REVIEW

File: tests/test_testing.py [hunk 1]
```diff
@@ -368,3 +369,34 @@ def test_httpclient_exception(pytester, book_list_html_response) -> None:
...
+
+
+def test_cli_rerun(pytester, book_list_html_response, capsys) -> None:
+    fixture = _save_fixture(
+        pytester,
+        page_cls=MyItemPage3,
+        page_inputs=[book_list_html_response],
+        expected={"foo": "bar2", "egg": "spam", "hello": "world"},
+    )
+    cli_main(["rerun", str(fixture.path)])
```
Comment (kmike) [inline tests/test_testing.py:386]: Initially I wanted to use subprocess, but had hard time making the tests/test_testing.MyItemPage importable. This way it works.

File: web_poet/testing/__main__.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+from web_poet.testing import Fixture
+
+
+def rerun(args):
+    fixture = Fixture(Path(args.fixture_path))
+    item = fixture.get_output()
+    if args.fields:
+        fields = args.fields.split(",")
+        item = {field: item[field] for field in fields}
```
Comment (BurnzZ) [inline web_poet/testing/__main__.py:12]: I think we can improve the error message here when users input a non-existing field. Or maybe ignore/supress the erorr?

**Comment by kmike** [comment]
Also not sure about `rerun` name, but haven't came up with a better name :( 

The good thing is that it's not a "scrapy savefixture rerun" command, but an unrelated `python -m web_poet.testing rerun`, so there is a bit less opportunity for confusion.

**Review by BurnzZ — APPROVED** [review]

File: web_poet/testing/__main__.py (hunk 2 above)
Comment (kmike) [reply web_poet/testing/__main__.py:12]: I was hoping nobody will notice it, and it'd be ok to just have a KeyError in the output :P

Comment (kmike) [reply web_poet/testing/__main__.py]: Done in https://github.com/scrapinghub/web-poet/pull/132/commits/c2b18065bd7837f0c3ea27257084a8b511cbe241.

**Comment by kmike** [comment]
Merging it, thanks @Gallaecio @BurnzZ!

**Comment by kmike** [comment]
Ouch, tests start to fail :)

**Comment by kmike** [comment]
Fixed in https://github.com/scrapinghub/web-poet/pull/137, it was unrelated.

**Pull request merged**